packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/widgets/toolbar.ts

```diff
@@ -74,27 +79,50 @@ function createCopyLinkToBlockMenuItem(
     ...item,
     action: (ctx: MenuContext) => {
       const baseUrl = getAffineCloudBaseUrl();
-      if (!baseUrl) return;
+      if (!baseUrl) {
+        ctx.close();
+        return;
+      }
 
       const { editor } = framework.get(EditorService);
       const mode = editor.mode$.value;
       const pageId = editor.doc.id;
       const workspaceId = editor.doc.workspace.id;
       const options: UseSharingUrl = { workspaceId, pageId, shareMode: mode };
+      let type = '';
 
       if (mode === 'page') {
         // maybe multiple blocks
         const blockIds = ctx.selectedBlockModels.map(model => model.id);
         options.blockIds = blockIds;
+        type = ctx.selectedBlockModels[0].flavour.split(':')[1];
```

Comment:
it means to remove the 'affine:' prefix right? I think it's ok to keep it

Reply:
yes, block/element's type is required